lib/elixir/lib/map.ex

```diff
@@ -636,6 +636,31 @@ defmodule Map do
     end
   end
 
+  @doc """
+  Returns and removes the value associated with `key` in `map` or raises
+  if `key` is not present.
+
+  Behaves the same as `pop/3` but raises if `key` is not present in `map`.
+
+  ## Examples
+
+      iex> Map.pop!(%{a: 1}, :a)
+      {1, %{}}
+      iex> Map.pop!(%{a: 1, b: 2}, :a)
+      {1, %{b: 2}}
+      iex> Map.pop!(%{a: 1}, :b)
+      ** (KeyError) key :b not found in: %{a: 1}
+
+  """
+  @doc since: "1.10.0"
+  @spec pop!(map, key) :: {value, map}
+  def pop!(map, key) do
+    case :maps.take(key, map) do
```

Comment:
Should we use the `:maps.take/2` function consistently in all the `pop`-style functions?

Reply:
We might but that's something for another day (and maybe someone who wants to do that? 😄)

Reply:
Done in master.